### PR TITLE
Add type hints to terraform utils and provider

### PR DESCRIPTION
**Pull request by gruebel**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Next batch of type hints, this time for `terrafom.checks.provider` and `terrafom.checks.utils`.

I also moved `convert_cloudformation_conf_to_iam_policy.py` to the cloudformation code base, which is probably the correct place to have it 😉 

Other than that, I also renamed `CustomAttributes` to `CustomTerraformAttributes` in the terraform code base to better differentiate the difference between the original one and the subclass.

### CODE REVIEW

File: checkov/terraform/checks/provider/base_check.py [hunk 1]
```diff
@@ -1,30 +1,22 @@
...
         raise NotImplementedError()
-
-    @classmethod
-    @scan_provider_conf.add_signature(args=["self", "conf"])
-    def _scan_provider_conf_self_conf(cls, wrapped):
-        def wrapper(self, conf, provider_type=None):
-            # keep default argument for entity_type so old code, that doesn't set it, will work.
-            return wrapped(self, conf)
-
-        return wrapper
```
Comment (nimrodkor) [inline checkov/terraform/checks/provider/base_check.py:30]: @gruebel Not sure we want to delete this... Do we?

File: checkov/terraform/graph_builder/graph_components/attribute_names.py [hunk 2]
```diff
@@ -4,10 +4,9 @@
 
 
 @dataclass
-class CustomAttributes(CustomAttributes):
-    SOURCE_MODULE = "source_module_"
+class CustomTerraformAttributes(CustomAttributes):
```
Comment (nimrodkor) [inline checkov/terraform/graph_builder/graph_components/attribute_names.py:7]: @gruebel Can we avoid this renaming?

**Review by nimrodkor — COMMENTED** [review]
Thanks @gruebel !

File: checkov/terraform/checks/provider/base_check.py (hunk 1 above)
Comment (gruebel) [reply checkov/terraform/checks/provider/base_check.py:30]: I checked current code base and there is no check, which utilises the old way of handling it with `provider_type`. if someone wants to also do something with the `provider_type`, then he/she can use the method `scan_entity_conf()` instead. Or is there any reason, why you want to keep it inside the code base?

File: checkov/terraform/graph_builder/graph_components/attribute_names.py (hunk 2 above)
Comment (gruebel) [reply checkov/terraform/graph_builder/graph_components/attribute_names.py:7]: let me try it again, I had some issues with it, that's why I changed it.

Comment (gruebel) [reply checkov/terraform/graph_builder/graph_components/attribute_names.py]: 😄 I found a way, please have a look into my new commit

**Review by nimrodkor — APPROVED** [review]

File: checkov/terraform/checks/provider/base_check.py (hunk 1 above)
Comment (nimrodkor) [reply checkov/terraform/checks/provider/base_check.py:30]: There is no check and I find it hard to think of a valid usecase. It seems more relevant for resource checks than provider checks... Let's remove it for now. Thanks!

**Pull request merged**